spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AboutController.java

```diff
@@ -315,4 +321,14 @@ else if (version.contains(".RC")) {
 		}
 		return result;
 	}
+
+	private Map getGitInfoFromEndpoints() {
```

Comment:
I think this method is still named from the original impl when you were using the endpoints. I think it may be more clear if we name it something like "addGitAndBuildInfoIfAvailable" .. that way we only call the setter if there was actually info available. 

```java
	private void addGitAndBuildInfoIfAvailable(AboutResource aboutResource) {
		Info.Builder builder = new Info.Builder();
		gitInfoContributor.ifAvailable(c -> c.contribute(builder));
		buildInfoContributor.ifAvailable(c -> c.contribute(builder));
		Map<String, Object> details = builder.build().getDetails();
		if (!ObjectUtils.isEmpty(details)) {
			aboutResource.setGitAndBuildInfo(details);
		}
	}
```

Reply:
Nice! done!